module-system/sov-state/src/codec.rs

```diff
@@ -154,3 +185,11 @@ where
         self.value_codec.encode_value(value)
     }
 }
+
+impl<K, Q, KC, VC> StateKeyEncodePreservingBorrow<K, Q> for PairOfCodecs<KC, VC>
+where
+    Q: ?Sized,
+    K: Borrow<Q>,
+    KC: StateKeyEncodePreservingBorrow<K, Q>,
```

Comment:
@neysofu can `StateKeyEncodePreservingBorrow` and `Borrow` be merged into single trait?

Reply:
I don't think we can, unfortunately. I remember trying. But there is a valid alternative: drop altogether all references to `Borrow` and decouple `StateKeyEncodePreservingBorrow` from the borrowing concept entirely. We'd end up with something akin to `StateKeyEquivalence`. Less type bounds as well (`K: Borrow<Q>` would get dropped), but the API is not as close to `HashMap`. @vlopes11 WDYT? I think I'd prefer that.